presto-hive/src/main/java/com/facebook/presto/hive/HivePartitionManager.java

```diff
@@ -263,7 +261,13 @@ public HivePartitionResult getPartitions(SemiTransactionalHiveMetastore metastor
 
         List<HiveColumnHandle> partitionColumns = getPartitionKeyColumnHandles(table);
 
-        List<HivePartition> partitions = getPartitionsAsList(getPartitionsIterator(metastore, tableHandle, constraint, session).iterator());
+        List<HivePartition> partitions = getPartitionsList(metastore, tableHandle, constraint, session);
+        if (partitions.size() > maxPartitionsPerScan) {
+            throw new PrestoException(HIVE_EXCEEDED_PARTITION_LIMIT, format(
+                    "Query over table '%s' can potentially read more than %s partitions",
+                    hiveTableHandle.getSchemaTableName().toString(),
+                    maxPartitionsPerScan));
+        }
```

Comment:
don't we want to fail fast if we exceed maxPartitionsPerScan .
can we have a counter (Atomic) to keep track of count from getPartitionListFromPartitionNames and fail if it is more than maxPartitionsPerScan

Reply:
As per our offline discussion, the suggestion was to do the check in the threads itself when they are doing the work parallely. The maxPartitionPerScan limit check should still happen outside of the parallel value parsing threads because not all users will enable this parallel-parsing-of-values feature. Also the number of queries hitting this limit will be few. So hence we decided to proceed with the code changes in the pr.